api/scripts/timeLogging.js

```diff
@@ -19,21 +19,29 @@ module.exports = (() => {
     }
 
     const addTimeEntries = (params) => {
-        return Promise.all(params.timeEntries.map(async t => {
-            if (!(await matchTimeEntry(t))) {
-                const contributor = await matchContributor(t)
+        let contributor
+        return params.timeEntries.map(async timeEntry => {
+            if (!(await matchTimeEntry(timeEntry))) {
+                contributor = await matchContributor(timeEntry)
```

Comment:
delete `let contributor` line
and on line 25 do `const contributor = await matchContributor(timeEntry)`